test/integration/StreamingHttpChannel2023.test.ts

```diff
@@ -0,0 +1,346 @@
+  getTestConfigPath,
+  getTestFolder,
+  instantiateFromConfig,
+  removeFolder,
+} from './Config';
+import namedNode = DataFactory.namedNode;
+
+const port = getPort('StreamingHTTPChannel2023');
+const baseUrl = `http://localhost:${port}/`;
+
+const rootFilePath = getTestFolder('StreamingHTTPChannel2023');
+const stores: [string, any][] = [
+  [ 'in-memory storage', {
+    configs: [ 'storage/backend/memory.json', 'util/resource-locker/memory.json' ],
+    teardown: jest.fn(),
+  }],
+  [ 'on-disk storage', {
+    configs: [ 'storage/backend/file.json', 'util/resource-locker/file.json' ],
+    teardown: async(): Promise<void> => removeFolder(rootFilePath),
+  }],
+];
+
+async function readChunk(reader: ReadableStreamDefaultReader): Promise<Store> {
+  const decoder = new TextDecoder();
+  const parser = new Parser();
+  const { value } = await reader.read();
+  const notification = decoder.decode(value);
+  return new Store(parser.parse(notification));
+}
+
+describe.each(stores)('A server supporting StreamingHTTPChannel2023 using %s', (name, { configs, teardown }): void => {
+  let app: App;
+  let store: ResourceStore;
+  const webId = 'http://example.com/card/#me';
+  const topic = joinUrl(baseUrl, '/foo');
+  const pathPrefix = '.notifications/StreamingHTTPChannel2023';
+  const receiveFrom = joinUrl(baseUrl, pathPrefix, '/foo');
+
+  beforeAll(async(): Promise<void> => {
+    const variables = {
+      ...getDefaultVariables(port, baseUrl),
+      'urn:solid-server:default:variable:rootFilePath': rootFilePath,
+    };
+
+    // Create and start the server
+    const instances = await instantiateFromConfig(
+      'urn:solid-server:test:Instances',
+      [
+        ...configs.map(getPresetConfigPath),
+        getTestConfigPath('streaming-http-notifications.json'),
+      ],
+      variables,
+    ) as Record<string, any>;
+    ({ app, store } = instances);
+
+    await app.start();
+  });
+
+  afterAll(async(): Promise<void> => {
+    await teardown();
+    await app.stop();
+  });
+
+  it('advertises streaming http endpoint in Link header.', async(): Promise<void> => {
+    await store.setRepresentation({ path: topic }, new BasicRepresentation('new', 'text/plain'));
+    const response = await fetch(topic);
+    expect(response.status).toBe(200);
+    const linkHeader = response.headers.get('link');
+    const match = /<([^>]+)>; rel="http:\/\/www\.w3\.org\/ns\/solid\/terms#updatesViaStreamingHttp2023"/u
+      .exec(linkHeader!);
+    expect(match![1]).toEqual(receiveFrom);
+  });
+
+  it('only allows GET on receiveFrom endpoint.', async(): Promise<void> => {
+    const methods = [ 'HEAD', 'PUT', 'POST' ];
+    for (const method of methods) {
+      const response = await fetch(receiveFrom, {
+        method,
+      });
+      expect(response.status).toBe(405);
+    }
+
+    // For some reason it differs
+    const del = await fetch(receiveFrom, {
+      method: 'DELETE',
+    });
+    expect(del.status).toBe(404);
```

Comment:
I don't know why `DELETE` responds with `404` instead of `405`, `OperationRouteHandler` has `"allowedMethods": [ "GET" ]`

Reply:
The problem is that the `RouterHandler` does all its checks in the `canHandle` call instead of the `handle` call. Which means that if it throws an error the server will just try the next handlers until it finds one that can handle it. The last one, the LDP handler, can always handle a request and will then throw a 404 because it can't find that resource.

There is indeed something to be said for the fact that method checks should be done in the `handle` call instead so a 405 is returned.

Reply:
And why does only the `DELETE` method respond with `404`, while `HEAD,` `PUT,` and `POST` all respond with `405`?

Reply:
> And why does only the `DELETE` method respond with `404`, while `HEAD,` `PUT,` and `POST` all respond with `405`?

I have no idea tbh, would be something to investigate 😅

Reply:
If you like, I could create an issue to track it after this PR gets merged.

Reply:
Sure